core/integration/container_test.go

```diff
@@ -4404,9 +4404,11 @@ func (ContainerSuite) TestNestedExec(ctx context.Context, t *testctx.T) {
 		output1a := runCtrs(c1, hostDir1, subdirA)
 		// run an exec that has /tmpdir/b/f included
 		output1b := runCtrs(c1, hostDir1, subdirB)
+
+		// TODO: fix comment
 		// sanity check: those should be different execs, *not* cached
 		// (this happens because content-hashing doesn't occur on the root dir)
-		require.NotEqual(t, output1a, output1b)
+		require.Equal(t, output1a, output1b)
```

Comment:
I'm guessing these are getting cached at the dagql layer now? Not at the buildkit layer?

But yes, this seems very reasonable :tada:

Reply:
Yes exactly, the caching actually improves in cases like this now 🚀